internal/kubernetes/kubernetes.go

```diff
@@ -189,3 +191,19 @@ func (k *Kubernetes) PatchConfigMap(name string, namespace string, data map[stri
 	})
 	return result, err
 }
+
+// GetSecretKeyValue will retrieve a particular data key from a secret
+func (k *Kubernetes) GetSecretKeyValue(secretName string, namespace string, dataKey string) (string, error) {
+	var err error
+	var secret *v1.Secret
+	var result string
+	err = retry.OnError(retry.DefaultBackoff, k.IsRetryError, func() error {
+		secret, err = k.client.CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		result = strings.TrimSpace(string(secret.Data[dataKey]))
```

Comment:
wont this cause a panic if secret.Data does't have a key by that name? We don't really do any validation that dataKey exists within the object.

Reply:
`secret.Data[dataKey]` should actually return the "zero" value for the type if `dataKey` is not found in the map, an empty string in this case. Instead of a panic, just a product of what Go is doing when it can't find a key. I did add an extra test per this point though https://github.com/Cray-HPE/loftsman/pull/31/commits/6f953e76fe4512b1098d22eee5ea4f6b44b0be8d. It's a really good catch and question.